diff --git a/src/itemlistformaction.cpp b/src/itemlistformaction.cpp
index 78570892..dd0a57e3 100644
--- a/src/itemlistformaction.cpp
+++ b/src/itemlistformaction.cpp
@@ -260,12 +260,8 @@ bool ItemListFormAction::process_operation(Operation op,
 						_("Error while toggling read flag: %s"),
 						e.what()));
 			}
-			if (!cfg->get_configvalue_as_bool(
+			if (cfg->get_configvalue_as_bool(
 					"toggleitemread-jumps-to-next-unread")) {
-				if (itempos < visible_items.size() - 1) {
-					list.set_position(itempos + 1);
-				}
-			} else {
 				process_operation(OP_NEXTUNREAD);
 			}
 			invalidate(itempos);